src/main/java/org/reficio/p2/TychoFeatureBuilder.java

```diff
@@ -0,0 +1,95 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.reficio.p2;
+
+import static org.twdata.maven.mojoexecutor.MojoExecutor.artifactId;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.configuration;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.element;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.executeMojo;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.executionEnvironment;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.goal;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.groupId;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.name;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.plugin;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.version;
+
+import java.io.File;
+import java.io.IOException;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.BuildPluginManager;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.tycho.core.TychoConstants;
+import org.eclipse.tycho.core.osgitools.targetplatform.DefaultDependencyArtifacts;
+
+public class TychoFeatureBuilder {
+    private static final String TYCHO_VERSION = "0.18.1";
+
+    File featureFile;
+    private final String outputDirectory;
+    String featureId;
+    String featureVersion;
+    private final MavenProject mavenProject;
+    private final MavenSession mavenSession;
+    private final BuildPluginManager buildPluginManager;
+
+    public TychoFeatureBuilder(File featureFile, String outputDirectory, String featureId, String featureVersion,
+                           MavenProject mavenProject, MavenSession mavenSession, BuildPluginManager buildPluginManager) {
+        this.featureFile = featureFile;
+        this.outputDirectory = outputDirectory;
+        this.featureId = featureId;
+        this.featureVersion = featureVersion;
+        this.mavenProject = mavenProject;
+        this.mavenSession = mavenSession;
+        this.buildPluginManager = buildPluginManager;
+    }
+
+    public void execute() throws MojoExecutionException, IOException {
+    	MavenProject mp =new MavenProject();
+    	mp.getModel().setGroupId(this.mavenProject.getGroupId());
+    	mp.getModel().setArtifactId(featureId);
+    	mp.getModel().setVersion(featureVersion);
+    	mp.getModel().setPackaging("eclipse-feature");
+    	mp.setPluginArtifactRepositories(this.mavenProject.getPluginArtifactRepositories());
+    	mp.setFile(featureFile); //sets the basedir for the MavenProject
+    	org.eclipse.tycho.artifacts.DependencyArtifacts da = new DefaultDependencyArtifacts();
+    	mp.setContextValue(TychoConstants.CTX_DEPENDENCY_ARTIFACTS, da);
+    	mavenSession.setCurrentProject(mp);
```

Comment:
Let's have a separate method where the MavenProject is prepared and returned from